examples/learning_graph_structured_data_h2.py

```diff
@@ -0,0 +1,172 @@
+"""Learning embedding of graph using Poincare Ball Model."""
+
+import logging
+
+import matplotlib.pyplot as plt
+
+import geomstats.backend as gs
+import geomstats.visualization as visualization
+from geomstats.datasets import graph_data_preparation as gdp
+from geomstats.geometry.hyperbolic import Hyperbolic
+
+
+def log_sigmoid(x):
+    """Logsigmoid function."""
+    return gs.log((1 / (1 + gs.exp(-x))))
+
+
+def grad_log_sigmoid(x):
+    """Gradient of logsigmoid function."""
+    return (1 / (1 + gs.exp(x)))
+
+
+def squared_distance(example_embedding, context_embedding,
+                     manifold):
+    """Squared distance considering the manifold metric."""
+    return \
+        manifold.metric.dist(example_embedding, context_embedding)**2
+
+
+def grad_squared_distance(example_embedding, context_embedding,
```

Comment:
This seems to be very close to what you want to do:
https://github.com/geomstats/geomstats/blob/05a2bcc2d696b56bc428c5d3d30e2be889920f4a/geomstats/geometry/riemannian_metric.py#L19

Is there anyway that you could *use* this function, or adding your function to hyperbolic.py if it very different?

Such functions are very useful even outside this example!

Reply:
Maybe renamed it in riemannian metric ? I do not know why this function is called loss instead of squaredDistance

Reply:
OK my next PR will be to refactor hyperbolic Space and Riemannian metrics.  I will make some changes to use  (and rename) upper-class methods.

Reply:
This function is called `loss` because it is used in a deep learning example on SE(3) in the `geomstats/applications` folder.